app/notification/model/notifier.py

```diff
@@ -24,65 +21,67 @@ def send_magic_link(json_data: dict, code: int = 200) -> tuple:
         expected contents to the user as expected by the
         govuk-notify-service template.
 
-        Raises error if any of the required  contents are incorrect
+        Raises HTTPError if any of the required  contents are incorrect
         or missing.
         """
         try:
-            data = MagicLink.from_json(json_data)
-            response = (
-                notifications_client.send_email_notification(
-                    email_address=data.contact_info,
-                    template_id=Config.MAGIC_LINK_TEMPLATE_ID,
-                    personalisation={
-                        "name of fund": data.fund_name,
-                        "link to application": data.magic_link,
-                        "request new link url": data.request_new_link_url,
-                        "contact details": data.contact_help_email,
-                    },
-                ),
-                code,
+            current_app.logger.info(
+                f"Connecting with {json_data.get('type')} service to process"
+                " contents"
+            )
+            data = Notification.process_data(json_data)
+            response = notifications_client.send_email_notification(
+                email_address=data.contact_info,
+                template_id=Config.MAGIC_LINK_TEMPLATE_ID,
+                personalisation={
+                    "name of fund": data.fund_name,
+                    "link to application": data.magic_link,
+                    "request new link url": data.request_new_link_url,
+                    "contact details": data.contact_help_email,
+                },
             )
+            current_app.logger.info("Contents have been mapped")
```

Comment:
Would a better log message be 'Call made to notify API'? As we've done more than just map some data by this point

Reply:
Yeah makes more sense. Done!